### PR TITLE
Convert inline admonishments to components

### PR DESCRIPTION
**Pull request by nik9000**
Converts inline admonishments into "component" style html using the
guidelines laid out here:
http://nicolasgallagher.com/about-html-semantics-front-end-architecture/

### CODE REVIEW

**Comment by nik9000** [comment]
This'll require a full `--rebuild` rebuild because it changes the xslt as well as the CSS.

**Comment by nik9000** [comment]
@elasticmachine, run elasticsearch-ci/docs rebuild

File: resources/website_common.xsl [hunk 1]
```diff
@@ -344,9 +344,9 @@
         <xsl:when test="attribute::revisionflag='deleted'">deprecated</xsl:when>
       </xsl:choose>
     </xsl:variable>
-    <span class="{$classname}">
-      [<span class="version"><xsl:value-of select="attribute::revision" /></span>]
-      <span class="detail">
+    <span class="Admonishment Admonishment--{$classname}">
+      [<span class="Admonishment-version Admonishment-version--{$classname}"><xsl:value-of select="attribute::revision" /></span>]
+      <span class="Admonishment-detail">
```
Comment (haywoood) [inline resources/website_common.xsl]: Nice!

File: resources/web/style/admonishment.pcss [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    .Admonishment-detail {
+      z-index: 10000;
+      left: -5em;
+      opacity: 1;
+      color: #555;
+      transition: color 0.5s ease;
+    }
+  }
+
+  &-title, &-version {
```
Comment (haywoood) [inline resources/web/style/admonishment.pcss]: Oh, that's cool!

Comment (nik9000) [reply resources/web/style/admonishment.pcss]: I like it! Nice and simple way to contain the component. It doesn't *always* work as you can see inside the style that is inside the hover.

File: resources/web/style/admonishment.pcss [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    font-family:Monaco, 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', 'Lucida Console', monospace!important;
```
Comment (haywoood) [inline resources/web/style/admonishment.pcss]: Just something to think about, generalized classes like maybe `u-monotype` or something that does the font-family and font-size may be helpful. We could end up with a bucket of general classes that can be combined for the effects we want in templates.

Nothing actionable, just something to think about when looking at CSS definitions.

Comment (nik9000) [reply resources/web/style/admonishment.pcss]: ++

File: resources/web/style/admonishment.pcss [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    font-size: 0.8em;
+    border-bottom: 1px dashed #00a9e5;
+  }
+  &-version--deprecated {
+    text-decoration: line-through;
```
Comment (haywoood) [inline resources/web/style/admonishment.pcss]: Continuing the line of thought above, what value does this have over a class like say, `u-strikethrough` or something.

Comment (nik9000) [reply resources/web/style/admonishment.pcss]: I don't think it has any beyond the "traditional" separation or style and markup. I'll see about adding those utilities before merging.

**Comment by nik9000** [comment]
@elasticmachine, run elasticsearch-ci/docs rebuild

**Comment by nik9000** [comment]
@haywoood, have a look now!

**Comment by nik9000** [comment]
@elasticmachine, run elasticsearch-ci/docs rebuild

**Comment by nik9000** [comment]
Thanks for reviewing @haywoood!

**Pull request merged**